kedro/io/data_catalog.py

```diff
@@ -567,16 +649,47 @@ def list(self, regex_search: str | None = None) -> list[str]:
             ) from exc
         return [dset_name for dset_name in self._data_sets if pattern.search(dset_name)]
 
+    def exists_in_catalog_config(self, dataset_name: str) -> bool:
+        """Check if a dataset exists in the catalog as an exact match or if it matches a pattern."""
+        if (
+            dataset_name in self._data_sets
+            or dataset_name in self._pattern_matches_cache
+        ):
+            return True
+        matched_pattern = self.match_name_against_patterns(dataset_name)
+        if matched_pattern:
+            # cache the "dataset_name -> pattern" match
+            self._pattern_matches_cache[dataset_name] = matched_pattern
+            return True
+        return False
```

Comment:
If we got rid of the overly verbose method name `self.match_name_against_patterns` and simply replace it with `resolve`, which could do all that's needed without leakage of abstraction, all this could be just `return self.resolve(dataset_name) is not None`.